op-batcher/batch_submitter.go

```diff
@@ -253,58 +254,65 @@ mainLoop:
 				l.log.Error("issue getting adding L2 Block", "err", err)
 				continue
 			}
-			data := new(bytes.Buffer)
-			data.WriteByte(derive.DerivationVersion0)
-			if err := l.ch.OutputFrame(data, l.cfg.MaxL1TxSize); err != nil {
-				l.log.Error("error outputting frame", "err", err)
-				continue
-			}
+			// Hand role do-while loop to fully pull all frames out of the channel
+			for {
+				// Collect the output frame
+				data := new(bytes.Buffer)
+				data.WriteByte(derive.DerivationVersion0)
+				done := false
+				if err := l.ch.OutputFrame(data, l.cfg.MaxL1TxSize); err == io.EOF {
+					done = true
+				} else if err != nil {
+					l.log.Error("error outputting frame", "err", err)
+					continue mainLoop
+				}
 
-			// Poll for new L1 Block and make a decision about what to do with the data
+				// Query for the submitter's current nonce.
+				walletAddr := crypto.PubkeyToAddress(l.cfg.PrivKey.PublicKey)
+				ctx, cancel = context.WithTimeout(l.ctx, time.Second*10)
+				nonce, err := l.cfg.L1Client.NonceAt(ctx, walletAddr, nil)
+				cancel()
+				if err != nil {
+					l.log.Error("unable to get current nonce", "err", err)
+					continue
```

Comment:
Same here. Maybe get the nonce before constructing the frame, so it can be a regular `continue` and we can retry getting the nonce.

Reply:
switch to `continue mainLoop`